src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/SystemModulesPlugin.java

```diff
@@ -1690,116 +1631,104 @@ void visitElement(T t, MethodVisitor mv) {
-                           allSystemModulesClassName,
-                           "<init>",
-                           "()V",
-                           false);
-        mv.visitInsn(ARETURN);
-        mv.visitMaxs(0, 0);
-        mv.visitEnd();
-
-        // defaultSystemModules()
-        mv = cw.visitMethod(ACC_STATIC,
-                            "defaultSystemModules",
-                            "()Ljdk/internal/module/SystemModules;",
-                            "()Ljdk/internal/module/SystemModules;",
-                            null);
-        mv.visitCode();
-        mv.visitTypeInsn(NEW, defaultSystemModulesClassName);
-        mv.visitInsn(DUP);
-        mv.visitMethodInsn(INVOKESPECIAL,
-                           defaultSystemModulesClassName,
-                           "<init>",
-                           "()V",
-                           false);
-        mv.visitInsn(ARETURN);
-        mv.visitMaxs(0, 0);
-        mv.visitEnd();
-
-        // moduleNames()
-        mv = cw.visitMethod(ACC_STATIC,
-                            "moduleNames",
-                            "()[Ljava/lang/String;",
-                            "()[Ljava/lang/String;",
-                            null);
-        mv.visitCode();
-        pushInt(mv, map.size());
-        mv.visitTypeInsn(ANEWARRAY, "java/lang/String");
-
-        int index = 0;
-        for (String moduleName : sorted(map.keySet())) {
-            mv.visitInsn(DUP);                  // arrayref
-            pushInt(mv, index);
-            mv.visitLdcInsn(moduleName);
-            mv.visitInsn(AASTORE);
-            index++;
-        }
-
-        mv.visitInsn(ARETURN);
-        mv.visitMaxs(0, 0);
-        mv.visitEnd();
-
-        // classNames()
-        mv = cw.visitMethod(ACC_STATIC,
-                            "classNames",
-                            "()[Ljava/lang/String;",
-                            "()[Ljava/lang/String;",
-                            null);
-        mv.visitCode();
-        pushInt(mv, map.size());
-        mv.visitTypeInsn(ANEWARRAY, "java/lang/String");
-
-        index = 0;
-        for (String className : sorted(map.values())) {
-            mv.visitInsn(DUP);                  // arrayref
-            pushInt(mv, index);
-            mv.visitLdcInsn(className.replace('/', '.'));
-            mv.visitInsn(AASTORE);
-            index++;
-        }
-
-        mv.visitInsn(ARETURN);
-        mv.visitMaxs(0, 0);
-        mv.visitEnd();
 
         // write the class file to the pool as a resource
-        String rn = "/java.base/" + SYSTEM_MODULES_MAP_CLASS + ".class";
-        ResourcePoolEntry e = ResourcePoolEntry.create(rn, cw.toByteArray());
+        String rn = "/java.base/" + SYSTEM_MODULES_MAP_CLASSNAME + ".class";
+        ResourcePoolEntry e = ResourcePoolEntry.create(rn, Classfile.build(
+                CD_SYSTEM_MODULES_MAP,
+                clb -> {
+                    clb.withFlags(ACC_FINAL+ACC_SUPER);
```

Comment:
Nit: consistent with line 578 with a space before and after `+`

Reply:
fixed, thanks.